changelogs/unreleased/gh-7486-deprecate-replication_synchro_timeout.md

```diff
@@ -0,0 +1,22 @@
+## feature/replication
+
+* Added a compat option `box_cfg_replication_synchro_timeout` to control
+  whether `replication_synchro_timeout` is accessible from `box.cfg`. It's
+  `old` by default, meaning the option is available and has the same default
+  value as before (5 seconds). If set to `new`, Tarantool will act as if
+  this option does not exist. The noticeable difference in new behavior is
+  that a synchronous transaction can remain in the synchro queue indefinitely
+  until it reaches a quorum of confirmations, just like canonical Raft
+  suggests.
+  By default, the option is `old` and will be switched to `new` in the next
+  major release. If you wish to try out the new behavior, you need to set the
+  option to `new` before the initial `box.cfg{}` call in your application.
```

Comment:
It seems one can alter the behavior on the fly, no?
No need to set the compat option before the first ever `box.cfg()` call, I mean.